### PR TITLE
Add build pipelines

### PR DESCRIPTION
**Pull request by zijchen**
Adds PR validation and a release pipeline

### CODE REVIEW

File: builds/template-steps-build-test.yml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+- task: DotNetCoreCLI@2
+  displayName: '.NET Build'
+  inputs:
+    command: build
+    projects: '${{ parameters.solution }}'
+    arguments: '--configuration ${{ parameters.configuration }} -p:GeneratePackageOnBuild=false'
+
+# CodeSign is only run on release builds
+- task: UseDotNet@2
+  displayName: 'Install .NET Core SDK 2.1.x for CodeSign'
```
Comment (kisantia) [inline builds/template-steps-build-test.yml:30]: CodeSign doesn't work with newer .NET Core SDKs?

**Review by kisantia — APPROVED** [review]

File: builds/template-steps-build-test.yml (hunk 1 above)
Comment (zijchen) [reply builds/template-steps-build-test.yml]: Looks like it's supported to .NET 5 SDK now, updated

File: builds/build-release.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  versionPatch: $[counter(variables['versionMinor'], 0)] # This will reset when we bump minor version
+  nugetVersion: '$(versionMajor).$(versionMinor).$(versionPatch)'
+
+stages:
+- stage: BuildPublish
+  displayName: 'Release Build and Publish'
+
+  jobs:
+  - job: BuildTestPublish
+    displayName: 'Release: Build, Test and Publish on Windows'
```
Comment (Benjin) [inline builds/build-release.yml:27]: No oxford comma? 😱

File: test/Microsoft.Buld.Sql.Tests/TestUtils.cs [hunk 3]
```diff
@@ -9,27 +9,38 @@ namespace Microsoft.Build.Sql.Tests
...
+                    dotnetPath = @"C:\Program Files\dotnet";
+                }
+                else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+                {
+                    dotnetPath = "/usr/bin/dotnet";
+                }
+                else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+                {
+                    dotnetPath = "/usr/local/share/dotnet";
+                }
```
Comment (Benjin) [inline test/Microsoft.Buld.Sql.Tests/TestUtils.cs:36]: I'm surprised there isn't a way built into dotnet to get this info, but I can't find anything about it!

**Review by Benjin — APPROVED** [review]

**Pull request merged**